src/monattack.cpp

```diff
@@ -3560,6 +3560,85 @@ void mattack::bite(monster *z, int index)
     target->check_dead_state();
 }
 
+void mattack::stretch_bite(monster *z, int index)
+{
+    if( !z->can_act() ) {
+        return;
+    }
+
+    // Let it be used on non-player creatures
+    // can be used at close range too!
+    Creature *target = z->attack_target();
+    if( target == nullptr || rl_dist( z->pos(), target->pos() ) > 3 ) {
```

Comment:
Also needs a check for clear line here or else it'll bite through solid glass and iron bars.